### PR TITLE
fix: template card cleanup

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### <samp>🤖 Generated by Copilot at 9f85663</samp>

This pull request improves the user interface of the journey template gallery and sections in the journeys-admin app. It enhances the style and functionality of the `TemplateGalleryCard`, `NavButton`, and `TemplateSection` components, and makes use of the `Box` and `SxProps` utilities from MUI.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/34356579/todos/6666672478)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] it should truncate title and add ellipsis if longer than three lines in template gallery card
- [ ] it should not have any hover stylings on template gallery card
- [ ] it should align the nav buttons to the image as opposed to bottom of card

# Walkthrough

### <samp>🤖 Generated by Copilot at 9f85663</samp>

*  Make the focus highlight transparent for the template gallery cards ([link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-a5682122bdc264ccbf7a71916c34a3cbf03eece7e2f730eb81fc9a1896ab0288L57-R64))
*  Limit the number of lines and add ellipsis for the journey titles on the cards ([link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-a5682122bdc264ccbf7a71916c34a3cbf03eece7e2f730eb81fc9a1896ab0288L116-R167))
*  Format the style prop of the card content for readability ([link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-a5682122bdc264ccbf7a71916c34a3cbf03eece7e2f730eb81fc9a1896ab0288L101-R113))
*  Add an optional sx prop to the nav button component to allow custom styles ([link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-1fabba564d7c1ee893cf1cbc5e0aac32e637a2c5b0a030384226b3bcd56272e4L2-R2), [link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-1fabba564d7c1ee893cf1cbc5e0aac32e637a2c5b0a030384226b3bcd56272e4L11-R15), [link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-1fabba564d7c1ee893cf1cbc5e0aac32e637a2c5b0a030384226b3bcd56272e4L37-R38))
*  Replace the stack component with a box component in the template section component to fix spacing and alignment issues ([link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-0016fcb7bad744aecc9ef2abfad388fd2d490ec024dc331c2800e4f4bd2069b0L1-R3), [link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-0016fcb7bad744aecc9ef2abfad388fd2d490ec024dc331c2800e4f4bd2069b0L64-R64))
*  Adjust the margin-top styles of the swiper and the nav buttons in the template section component ([link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-0016fcb7bad744aecc9ef2abfad388fd2d490ec024dc331c2800e4f4bd2069b0L104-R104), [link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-0016fcb7bad744aecc9ef2abfad388fd2d490ec024dc331c2800e4f4bd2069b0L116-R128))
*  Remove the margin-top style from the nav button component ([link](https://github.com/JesusFilm/core/pull/1982/files?diff=unified&w=0#diff-1fabba564d7c1ee893cf1cbc5e0aac32e637a2c5b0a030384226b3bcd56272e4L28))

### CODE REVIEW

**Review by Kneesal — COMMENTED** [review]
image


https://caniuse.com/?search=line%20clamp

^ in regards to using webkit line clamp

File: apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx [hunk 1]
```diff
@@ -54,7 +54,14 @@ export function TemplateGalleryCard({
         passHref
         legacyBehavior
       >
-        <CardActionArea sx={{ height: 'inherit' }}>
+        <CardActionArea
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:57]: If we're going to remove the focusHighlight, why do we even need this CardActionArea wrapper?

File: apps/journeys-admin/src/components/TemplateSections/TemplateSection/TemplateSection.tsx [hunk 2]
```diff
@@ -113,8 +113,18 @@ export function TemplateSection({
...
         </Swiper>
       )}
-      <NavButton variant="prev" ref={prevRef} disabled={journeys == null} />
-      <NavButton variant="next" ref={nextRef} disabled={journeys == null} />
-    </Stack>
+      <NavButton
+        variant="prev"
+        ref={prevRef}
+        disabled={journeys == null}
+        sx={{ mt: { lg: '-19%', xl: '-17%' } }}
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateSections/TemplateSection/TemplateSection.tsx:120]: Why pass this in if it's the same for both nav buttons and this NavButton is specific to this TemplateSection? Just add the sx to `NavButton` ?

File: apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx [hunk 3]
```diff
@@ -113,27 +125,46 @@ export function TemplateGalleryCard({
                 >
                   {date} ● {displayLanguage}
                 </Typography>
-                <Typography
-                  variant="subtitle2"
+                <Box
                   sx={{
-                    my: 1,
-                    display: { xs: 'none', lg: 'block' }
+                    display: { xs: 'none', lg: '-webkit-box' },
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:130]: -webkit-box is very rarely used, what are you trying to achieve? Hard to understand from basecamp descriptions with no images...

Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:130]: see comment: 
image


truncating the title of the template gallery card to three lines and adding an ellipsis at the end

**Review by storyworks — APPROVED** [review]

File: apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx [hunk 4]
```diff
@@ -100,7 +99,7 @@ export function TemplateGalleryCard({
               }}
             />
           )}
-          <CardContent
+          <Box
             sx={{
               display: 'flex',
```
Comment (storyworks) [inline apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:104]: Stack....

Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:104]: bruhhh my bad

File: apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.spec.tsx [hunk 5]
```diff
@@ -92,8 +92,12 @@ describe('TemplateGalleryCard', () => {
     expect(getByText('Aug, 2023 ● Kalagan (TK)')).toBeInTheDocument()
   })
 
-  it('should link to template details', () => {
-    const { getByRole } = render(<TemplateGalleryCard journey={journey} />)
-    expect(getByRole('link')).toHaveAttribute('href', '/templates/template-id')
+  it('should link to template details', async () => {
+    const { container } = render(<TemplateGalleryCard journey={journey} />)
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.spec.tsx:96]: Any other way we can try do that same test but without using container?

**Review by GeronimoJohn — APPROVED** [review]

File: apps/journeys-admin/src/components/TemplateSections/TemplateSection/TemplateSection.tsx [hunk 6]
```diff
@@ -115,7 +115,11 @@ export function TemplateSection({
           observer
           observeParents
           breakpoints={swiperBreakpoints}
-          style={{ overflow: 'visible' }}
+          navigation={{
+            nextEl: nextRef.current,
+            prevEl: prevRef.current
+          }}
```
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateSections/TemplateSection/TemplateSection.tsx:121]: must have come in during a merge conflict

File: apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx [hunk 7]
```diff
@@ -113,27 +119,46 @@ export function TemplateGalleryCard({
...
                 >
-                  {journey.title}
-                </Typography>
-                <Typography
-                  variant="subtitle3"
+                  <Typography
+                    variant="subtitle2"
+                    sx={{
+                      my: 1,
+                      display: { xs: 'none', lg: 'block' }
```
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:136]: both typographies needed as they are different variants. subtitle 1 and subtitle 2

File: apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx [hunk 8]
```diff
@@ -113,27 +119,46 @@ export function TemplateGalleryCard({
...
+                    WebkitLineClamp: 3
                   }}
                 >
-                  {journey.title}
-                </Typography>
+                  <Typography
+                    variant="subtitle3"
+                    sx={{
+                      my: 1,
+                      display: { xs: 'block', lg: 'none' }
```
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:156]: both typographies needed as they are different variants. subtitle 1 and subtitle 2

File: apps/journeys-admin/src/components/TemplateSections/TemplateSection/TemplateSection.tsx [hunk 9]
```diff
@@ -115,7 +115,11 @@ export function TemplateSection({
...
+          style={{ overflow: 'visible', marginTop: 16 }}
```
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateSections/TemplateSection/TemplateSection.tsx:122]: image

the spacing 4 in the stack above set the spacings as per designs previously

[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateSections/TemplateSection/TemplateSection.tsx:122]: removing it means you have to manually set the spacing now

File: apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:136]: oh yep you're right, not needed

File: apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateGalleryCard/TemplateGalleryCard.tsx:156]: oh yep you're right, not needed

**Review by GeronimoJohn — APPROVED** [review]

File: apps/journeys-admin/src/components/ThemeProvider/admin/tokens/typography.ts [hunk 10]
```diff
@@ -78,7 +78,8 @@ export const adminTypography: Pick<ThemeOptions, 'typography'> = {
     subtitle3: {
       fontSize: 14,
       fontWeight: 600,
-      lineHeight: '19px'
+      lineHeight: '19px',
+      fontFamily: '"Montserrat", "Open Sans", "Tahoma", "Verdana", sans-serif'
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/ThemeProvider/admin/tokens/typography.ts:82]: see: https://3.basecamp.com/3105655/buckets/34356579/todos/6684358419#__recording_6691290741

File: apps/journeys-admin/src/components/ThemeProvider/admin/tokens/typography.ts [hunk 11]
```diff
@@ -104,7 +105,8 @@ export const adminTypography: Pick<ThemeOptions, 'typography'> = {
       fontSize: 10,
       lineHeight: '16px',
       letterSpacing: 1,
-      textTransform: 'uppercase'
+      textTransform: 'uppercase',
+      fontFamily: '"Montserrat", "Open Sans", "Tahoma", "Verdana", sans-serif'
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/ThemeProvider/admin/tokens/typography.ts:109]: see: https://3.basecamp.com/3105655/buckets/34356579/todos/6684358419#__recording_6691290741

**Review by GeronimoJohn — APPROVED** [review]

**Pull request merged**